File: client/types/event.ts
import { IUser } from ".";

export type IEvent = {
    id?: string;
    title: string;
    location?: string;
    time?: string;
    startDate: Date;
    endDate?: Date;
    userId?: string;
    user?: IUser;
};

export type IEventApiResponse = {
    id: string;
    title: string;
    location?: string;
    time?: string;
    startDate: string;
    endDate?: string;
    userId: string;
    user?: IUser;
}

File: server/Models/Event.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server.Models
{
    public class Event
    {
        [Key]
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string? Location { get; set; }
        public string? Time { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime? EndTime { get; set; }
        public Guid UserId { get; set; }

        [ForeignKey("UserId")]
        public User? User { get; set; }
    }
}
